Mathlib/Util/AssertExists.lean

```diff
@@ -22,6 +23,66 @@ Implement `assert_instance` and `assert_no_instance`
 section
 open Lean Elab Meta Command
 
+namespace Mathlib.AssertNotExist
+
+/-- `#check_assertions` retrieves all declarations and all imports that were declared
+not to exist so far (including in transitively imported files) and reports their current
+status:
+* ✓ means the declaration or import exists,
+* × means the declaration or import does not exist.
+
+This means that the expectation is that all checks *succeed* by the time `#check_assertions`
+is used, typically once all of `Mathlib` has been built.
+
+If all declarations and imports are available when `#check_assertions` is used,
+then the command logs an info. Otherwise, it emits a warning.
+
+The variant `#check_assertions!` only prints declarations/imports that are not present in the
+environment.  In particular, it is silent if everything is imported, making it useful for testing.
+-/
+elab "#check_assertions" tk:("!")?: command => do
+  let env ← getEnv
+  let ext := assertExistsExt.getState env
+  if ext.isEmpty && tk.isNone then logInfo "No assertions made." else
+  let allMods := env.allImportedModuleNames
+  let mut msgs := #[m!""]
+  let mut outcome := m!""
+  let mut (allExist?, cond) := (true, false)
+  for d in ext.toArray.qsort fun d e => (e.isDecl < d.isDecl) ||
+      (e.isDecl == d.isDecl && (d.givenName.toString < e.givenName.toString)) do
+    let type := if d.isDecl then "declaration" else "module"
+    if d.isDecl then
```

Comment:
This logic makes me think there should only be one command that handles both declarations and imports.

Reply:
Hmm the comment wasn't focused correctly. I mean the `if d.isDecl` branching

Reply:
If there happened to be collision, I would make sure both don't exist and let the error disambiguate if that failed. But this is not in scope here.